test/plots/barPlotTests.ts

```diff
@@ -1 +1,783 @@
+          baseScale.domain(["-A"]);
+          barPlot.addDataset(dataset);
+          barPlot.renderTo(svg);
+
+          assert.strictEqual(barPlot.content().selectAll("rect").size(), 0, "draws no bars when the domain contains no data points");
+        });
+
+        afterEach(function() {
+          if (this.currentTest.state === "passed") {
+            barPlot.destroy();
+            svg.remove();
+          }
+        });
+      });
+
+      describe(`autodomaining when ${orientation}`, () => {
+        let svg: d3.Selection<void>;
+        let baseScale: Plottable.Scales.Linear;
+        let valueScale: Plottable.Scales.Linear;
+        let barPlot: Plottable.Plots.Bar<number, number>;
+
+        const baseAccessor = (d: any) => d.base;
+        const valueAccessor = (d: any) => d.value;
+
+        beforeEach(() => {
+          svg = TestMethods.generateSVG();
+          barPlot = new Plottable.Plots.Bar<number, number>(orientation);
+          baseScale = new Plottable.Scales.Linear();
+          valueScale = new Plottable.Scales.Linear();
+          if (orientation === Plottable.Plots.Bar.ORIENTATION_VERTICAL) {
+            barPlot.x(baseAccessor, baseScale);
+            barPlot.y(valueAccessor, valueScale);
+          } else {
+            barPlot.y(baseAccessor, baseScale);
+            barPlot.x(valueAccessor, valueScale);
+          }
+        });
+
+        it("computes the base scale domain correctly when there is only one data point", () => {
+          const singlePointData = [
+            { base: baseScale.domain()[1] + 10, value: 5 }
+          ];
+          barPlot.addDataset(new Plottable.Dataset(singlePointData));
+          barPlot.renderTo(svg);
+          const baseScaleDomain = baseScale.domain();
+          assert.operator(baseScaleDomain[0], "<=", singlePointData[0].base, "lower end of base domain is less than the value");
+          assert.operator(baseScaleDomain[1], ">=", singlePointData[0].base, "upper end of base domain is greater than the value");
+        });
+
+        it("base scale domain does not change when autoDomain is called more than once", () => {
+          const data = [
+            { base: 0, value: 1 },
+            { base: 1, value: 2 }
+          ];
+          barPlot.addDataset(new Plottable.Dataset(data));
+          barPlot.renderTo(svg);
+
+          const baseDomainAfterRendering = baseScale.domain();
+          baseScale.autoDomain();
+          const baseDomainAfterAutoDomaining = baseScale.domain();
+          assert.deepEqual(baseDomainAfterRendering, baseDomainAfterAutoDomaining, "calling autoDomain() again does not change the domain");
+        });
+
+        afterEach(function() {
+          if (this.currentTest.state === "passed") {
+            barPlot.destroy();
+            svg.remove();
+          }
+        });
+      });
+
+      describe(`auto bar width calculation when ${orientation}`, () => {
+        const scaleTypes = ["Linear", "ModifiedLog", "Time"];
```

Comment:
Why not
`[Plottable.Scales.Linear, Plottable.Scales.ModifiedLog, Plottable.Scales.Time]`?